docker/daemon.go

```diff
@@ -150,16 +154,31 @@ func mainDaemon() {
 	job.Setenv("TlsCert", *flCert)
 	job.Setenv("TlsKey", *flKey)
 	job.SetenvBool("BufferRequests", true)
-	err := job.Run()
+
+	// The serve API job never exits unless an error occurs
+	// We need to start it as a goroutine and wait on it so
+	// daemon doesn't exit
+	serveAPIWait := make(chan struct{})
+	go func() {
+		defer close(serveAPIWait)
+
+		if err := job.Run(); err != nil {
+			log.Errorf("Error from ServeAPI: %v", err)
+		}
+	}()
 
 	// Wait for the daemon startup goroutine to finish
 	// This makes sure we can actually cleanly shutdown the daemon
-	log.Infof("waiting for daemon to initialize")
-	<-daemonWait
-	eng.Shutdown()
+	log.Debug("waiting for daemon to initialize")
+	err := <-daemonInitWait
 	if err != nil {
+		eng.Shutdown()
 		// log errors here so the log output looks more consistent
 		log.Fatalf("shutting down daemon due to errors: %v", err)
 	}
+	log.Info("Daemon has completed initialization")
```

Comment:
yeah, I changed to other to output on debug only, but it doesn't help debugging since the goroutine is done at that point--I'll remove it